src/spetlr/cache/CachedLoader.py

```diff
@@ -245,3 +264,51 @@ def _discard_non_new_rows_against_cache(
             )
 
         return result
+
+    def _perform_provisional_markup(self, df: DataFrame) -> int:
+        """The cache table is updated with a provisional update where all
+        *potentially* affected rows have their hash set to zero.
+        If the following operation succeeds, the provisional markup is rolled
+        back and the *actually* affected rows have their correct markup applied.
+        If the operation fails, however, no further corrective action is required
+        to ensure a full re-write of the row in the target system.
+
+        The function returns the version that should be restored if all goes to
+        plan."""
+        spark = Spark().get()
+        p = self.params
+
+        pre_version = (
+            Spark.get()
+            .sql(f"DESCRIBE HISTORY {p.cache_table_name} LIMIT 1")
```

Comment:
order by version?

Reply:
Odering not needed. The LIMIT 1 will only show the last version.